### DISCUSSION TITLE
Dockerize Airflow with a different python version than the one from docker image

### DISCUSSION THREAD

**Discussion by ferrannoguera** [apache/airflow#36074] [Q&A] [answered]
Hi everyone! I'm using [apache/airflow:2.6.2-python3.11](https://hub.docker.com/layers/apache/airflow/2.6.2-python3.11/images/sha256-fabfdc720f8457a2cd6779eb76d2a4fe4d8c1697a37659f017a041b44aa45ce1) official image to dockerize my airflow. The problem is that I would need to use python `3.11.3` and the default python version from the image is `3.11.4`.

I've tried using `PYTHON_BASE_IMAGE=python:3.11.3` but it did not work. Also tried manually installing python within the same dockerfile, but it gave me a lot of errors and for what I've seen `PYTHON_BASE_IMAGE` should do it. I could potentially upgrade the Airflow version to whichever is the newest one if that was an issue, but I strictly need python version to be 3.11.3. Any ideas?

Thanks a lot for your help!

**Comment by potiuk** [comment, marked as answer]
You can attempt (there is no guarantee it will work but should) to build your own custom image. You need to build it from scratch as described in https://airflow.apache.org/docs/docker-stack/build.html#customizing-the-image

Read the chapter to learn how to do it. 

You have some examples of how to build the image using different base image:

https://airflow.apache.org/docs/docker-stack/build.html#examples-of-image-customizing

```
export AIRFLOW_VERSION=2.3.4
export DOCKER_BUILDKIT=1

docker build . \
    --build-arg PYTHON_BASE_IMAGE="python:3.8-slim-bullseye" \
    --build-arg AIRFLOW_VERSION="${AIRFLOW_VERSION}" \
    --tag "my-pypi-selected-version:0.0.1"
```

You will need to use the right base image, and right python version (bullseye was the base image used in 2.6).

Just read and follow the docs
